client/lib/signup/step-actions.js

```diff
@@ -174,6 +174,14 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 			siteType ||
 			getSiteVertical( state );
 		newSiteParams.find_available_url = true;
+	} else if ( get( getSignupDependencyStore( state ), 'shouldHideFreePlan', false ) ) {
```

Comment:
Looks like the `newSiteParams` here are the same as for the `isDomainStepSkippable` block above — could these be combined, or did you want to keep them separate in case there are differences between them in the future?

Reply:
Addressed in [00f8cc2](https://github.com/Automattic/wp-calypso/pull/37554/commits/00f8cc226c263c8b4f42037c189f453ee734c181)